From: Lennart Poettering <lennart@poettering.net>
Date: Fri, 17 Jan 2020 15:09:01 +0100
Subject: mount: make checks on perpetual mount units more lax

We don#t really care where perpetual mounts are mounted from, since they
have to exist since before we run anyway.

(cherry picked from commit 0879fbd6fedc2f813aebcb1a4eba005a99525bde)
---
 src/core/mount.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/src/core/mount.c b/src/core/mount.c
index dfed691..8901893 100644
--- a/src/core/mount.c
+++ b/src/core/mount.c
@@ -537,10 +537,9 @@ static int mount_verify(Mount *m) {
         }
 
         p = get_mount_parameters_fragment(m);
-        if (p && !p->what) {
-                log_unit_error(UNIT(m), "What= setting is missing. Refusing.");
-                return -ENOEXEC;
-        }
+        if (p && !p->what && !UNIT(m)->perpetual)
+                return log_unit_error_errno(UNIT(m), SYNTHETIC_ERRNO(ENOEXEC),
+                                            "What= setting is missing. Refusing.");
 
         if (m->exec_context.pam_name && m->kill_context.kill_mode != KILL_CONTROL_GROUP) {
                 log_unit_error(UNIT(m), "Unit has PAM enabled. Kill mode must be set to control-group'. Refusing.");
